### ISSUE TITLE
[FATAL] trojan service (client) started at 127.0.0.1:1080

### ISSUE THREAD

**Issue by b1b2b3b4b5b6** [trojan-gfw/trojan#378]
- [x] I certify that I have read the contributing guidelines and I acknowledge if I don't follow the format below, or I'm using an old version of trojan, or I apparently fail to provide sufficient information (such as logs, specific numbers), or I don't check this box, my issue will be closed immediately without any notice.

**Trojan Version**
1.10.0

**Describe the bug**

run trojan with config.json in /etc/trojan/config, output is this:
`Welcome to trojan 1.10.0
[2020-04-05 23:27:21] [FATAL] trojan service (client) started at 127.0.0.1:1080`
and can not connect with sock5

**To Reproduce**

1. apt install trojan
2. copy my windows version 'config.json' file to /etc/trojan/config
3. edit "log_level": "4" to 0 
4. run trojan
5. see error

**Expected behavior**

**Logs**
`Welcome to trojan 1.10.0
[2020-04-05 23:27:21] [FATAL] trojan service (client) started at 127.0.0.1:1080`

**Environment**
阿里云ecs
Linux 4.19.0-8-amd64 #1 SMP Debian 4.19.98-1 (2020-01-26) x86_64

**Additional context**
{
    "local_addr": "127.0.0.1",
    "local_port": 1080,
    "log_level": "0",
    "password": [
        "password"
    ],
    "remote_addr": "os1-7.sstr-api.xyz",
    "remote_port": 443,
    "run_type": "client",
    "ssl": {
        "alpn": [
            "h2",
            "http/1.1"
        ],
        "cert": "",
        "cipher": "ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES128-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA:AES128-SHA:AES256-SHA:DES-CBC3-SHA",
        "cipher_tls13": "TLS_AES_128_GCM_SHA256:TLS_CHACHA20_POLY1305_SHA256:TLS_AES_256_GCM_SHA384",
        "curves": "",
        "reuse_session": false,
        "session_ticket": false,
        "sni": "",
        "verify": true,
        "verify_hostname": true
    },
    "tcp": {
        "fast_open": false,
        "fast_open_qlen": 20,
        "keep_alive": true,
        "no_delay": true,
        "reuse_port": false
    }
}

**Comment by GreaterFire** [comment]
This is not an error.

**Comment by LeeiFrankJaw** [comment]
> This is not an error.

If this is not an error, then you should explain why this is not an error.  After running into the same problem and googled into here, only to find out a closed issue with a "this is not error" reply.

**Comment by LeeiFrankJaw** [comment]
>     * [x]  I certify that I have read the contributing guidelines and I acknowledge if I don't follow the format below, or I'm using an old version of trojan, or I apparently fail to provide sufficient information (such as logs, specific numbers), or I don't check this box, my issue will be closed immediately without any notice.
> 
> 
> **Trojan Version**
> 1.10.0
> 
> **Describe the bug**
> 
> run trojan with config.json in /etc/trojan/config, output is this:
> `Welcome to trojan 1.10.0 [2020-04-05 23:27:21] [FATAL] trojan service (client) started at 127.0.0.1:1080`
> and can not connect with sock5
> 
> **To Reproduce**
> Steps to reproduce the behavior:
> 
>     1. apt install trojan
> 
>     2. copy my windows version 'config.json' file to /etc/trojan/config
> 
>     3. edit "log_level": "4" to 0
> 
>     4. run trojan
> 
>     5. see error
> 
> 
> **Expected behavior**
> A clear and concise description of what you expected to happen.
> 
> **Logs**
> `Welcome to trojan 1.10.0 [2020-04-05 23:27:21] [FATAL] trojan service (client) started at 127.0.0.1:1080`
> 
> **Environment**
> 阿里云ecs
> Linux 4.19.0-8-amd64 #1 SMP Debian 4.19.98-1 (2020-01-26) x86_64
> 
> **Additional context**
> {
> "local_addr": "127.0.0.1",
> "local_port": 1080,
> "log_level": "0",
> "password": [
> "password"
> ],
> "remote_addr": "os1-7.sstr-api.xyz",
> "remote_port": 443,
> "run_type": "client",
> "ssl": {
> "alpn": [
> "h2",
> "http/1.1"
> ],
> "cert": "",
> "cipher": "ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES128-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA:AES128-SHA:AES256-SHA:DES-CBC3-SHA",
> "cipher_tls13": "TLS_AES_128_GCM_SHA256:TLS_CHACHA20_POLY1305_SHA256:TLS_AES_256_GCM_SHA384",
> "curves": "",
> "reuse_session": false,
> "session_ticket": false,
> "sni": "",
> "verify": true,
> "verify_hostname": true
> },
> "tcp": {
> "fast_open": false,
> "fast_open_qlen": 20,
> "keep_alive": true,
> "no_delay": true,
> "reuse_port": false
> }
> }

I encounter the same problem today.  But in the end I managed to solve my problem. 

You know what.  The culprit is exactly the trojan binary itself.  And you have to build the trojan yourself.  The trojan package from Debian repository is broken for unknown reasons.  It might be due to missing dependencies or outdated trojan version. The version from the repository is 1.10.0 with boost 1.67.  I just build version 1.14.1 from source with boost 1.72.

The problematic trojan package is the following.

```
Package: trojan
Version: 1.10.0-3
Priority: optional
Section: web
Maintainer: GreaterFire <GreaterFire@protonmail.com>
Installed-Size: 614 kB
Depends: libboost-program-options1.67.0, libboost-system1.67.0, libc6 (>= 2.14), libgcc1 (>= 1:3.0), libmariadb3 (>= 3.0.0), libssl1.1 (>= 1.1.1), libstdc++6 (>= 5.2)
Recommends: ca-certificates
Suggests: default-mysql-server
Homepage: https://github.com/trojan-gfw/trojan
Download-Size: 185 kB
Description: fast and lightweight proxy that helps you bypass firewalls
 Trojan features multiple protocols over TLS to avoid both active/passive
 detections and ISP QoS limitations. It tries to hide a proxy service
 behind a normal HTTPS service. Trojan claims to be unidentifiable.

N: There is 1 additional record. Please use the '-a' switch to see it
```

**Comment by Chigusa0w0** [comment]
@LeeiFrankJaw Read the message and understand what it means.
This could help you understand what's happening when the message pops up.
https://github.com/trojan-gfw/trojan/blob/fe1f297d12c85e4889a70c94469b5ced9be4d8a0/src/service.cpp#L204
And this is the "fix" https://github.com/trojan-gfw/trojan/commit/246adc421e3a31c16dd20fe5abd3beb9b249e6dd

**Comment by LeeiFrankJaw** [comment]
> @LeeiFrankJaw Read the message and understand what it means.
> This could help you understand what's happening when the message pops up.
> 
> https://github.com/trojan-gfw/trojan/blob/fe1f297d12c85e4889a70c94469b5ced9be4d8a0/src/service.cpp#L204
> 
> And this is the "fix" [246adc4](https://github.com/trojan-gfw/trojan/commit/246adc421e3a31c16dd20fe5abd3beb9b249e6dd)

No. Maybe this is not what I mean in the original place.  The first message is nothing for me.  The following messages for me are the following.

```
Welcome to trojan 1.10.0
[2020-06-02 08:55:47] [FATAL] trojan service (client) started at 127.0.0.1:1080
[2020-06-02 10:35:26] [ERROR] 172.19.77.133:38448 unknown protocol
[2020-06-02 10:35:26] [INFO] 172.19.77.133:38448 disconnected, 0 bytes received, 0 bytes sent, lasted for 0 seconds
[2020-06-02 10:35:26] [ERROR] 172.19.77.133:44350 unknown protocol
[2020-06-02 10:35:26] [INFO] 172.19.77.133:44350 disconnected, 0 bytes received, 0 bytes sent, lasted for 0 seconds
[2020-06-02 10:35:27] [ERROR] 172.19.77.133:38278 unknown protocol
[2020-06-02 10:35:27] [INFO] 172.19.77.133:38278 disconnected, 0 bytes received, 0 bytes sent, lasted for 0 seconds
[2020-06-02 10:35:28] [ERROR] 172.19.77.133:59106 unknown protocol
[2020-06-02 10:35:28] [INFO] 172.19.77.133:59106 disconnected, 0 bytes received, 0 bytes sent, lasted for 0 seconds
[2020-06-02 10:35:28] [ERROR] 172.19.77.133:59108 unknown protocol
[2020-06-02 10:35:28] [INFO] 172.19.77.133:59108 disconnected, 0 bytes received, 0 bytes sent, lasted for 0 seconds
[2020-06-02 10:35:30] [ERROR] 172.19.77.133:58216 unknown protocol
[2020-06-02 10:35:30] [INFO] 172.19.77.133:58216 disconnected, 0 bytes received, 0 bytes sent, lasted for 0 seconds
[2020-06-02 10:35:31] [ERROR] 172.19.77.133:34862 unknown protocol
[2020-06-02 10:35:31] [INFO] 172.19.77.133:34862 disconnected, 0 bytes received, 0 bytes sent, lasted for 0 seconds
[2020-06-02 10:35:31] [ERROR] 172.19.77.133:36652 unknown protocol
[2020-06-02 10:35:31] [INFO] 172.19.77.133:36652 disconnected, 0 bytes received, 0 bytes sent, lasted for 0 seconds
[2020-06-02 10:36:16] [ERROR] 172.19.77.133:39636 unknown protocol
[2020-06-02 10:36:16] [INFO] 172.19.77.133:39636 disconnected, 0 bytes received, 0 bytes sent, lasted for 0 seconds
[2020-06-02 10:36:16] [ERROR] 172.19.77.133:39248 unknown protocol
[2020-06-02 10:36:16] [INFO] 172.19.77.133:39248 disconnected, 0 bytes received, 0 bytes sent, lasted for 0 seconds
[2020-06-02 10:36:53] [ERROR] 172.19.77.133:39428 unknown protocol
[2020-06-02 10:36:53] [INFO] 172.19.77.133:39428 disconnected, 0 bytes received, 0 bytes sent, lasted for 0 seconds
[2020-06-02 10:36:54] [ERROR] 172.19.77.133:58854 unknown protocol
[2020-06-02 10:36:54] [INFO] 172.19.77.133:58854 disconnected, 0 bytes received, 0 bytes sent, lasted for 0 seconds
[2020-06-02 10:36:54] [FATAL] trojan service stopped
```

The newer version does solve my real connection problem. It might be the case my problem is really different from the original poster, which might be just an improper wording choice.  But for me, I simply cannot connect.
